### ISSUE TITLE
Exception on numbers over signed int32 max value

### ISSUE THREAD

**Issue by ike709** [wixoaGit/OpenDream#32]
Example code:
```
#define BIGNUM 4E9

/proc/is_small_number(x)
    if(x < BIGNUM)
        return 1
    return 0
```
Error:
```
Unhandled exception. System.Exception: Expected ')'
   at OpenDreamShared.Compiler.Parser`1.Consume(TokenType type, String errorMessage) in C:\Users\Ike\Documents\GitHub\OpenDream\OpenDream\OpenDreamShared\Compiler\Parser.cs:line 67
[...]
```

**Comment by ike709** [comment]
`4e9` is 4 billion while C#'s `int` is from –2,147,483,648 to 2,147,483,647

**Comment by wixoaGit** [comment]
This was fixed in aeee0abffcb9f60e5b6a99453f5a3d133648dbbc

The lexer would previously only parse a number as a float if it was `1.#INF`, or contained a period or lowercase `e`. It now always tries to read it as an integer, and falls back to a float if it can't.
